python/benchmarks/coco/datagen.py

```diff
@@ -33,21 +36,14 @@ def _get_instances_json(self, split):
             return json.load(fobj)
 
     def _instances_to_df(self, split, instances_json):
-        df = pd.DataFrame(instances_json["annotations"])
-        df["segmentation"] = df.segmentation.apply(_convert_segmentation)
-        df["iscrowd"] = df.iscrowd.astype("bool")
-        # Convert coco dataset bounding box [x,y,width,height] to [x0,y0,x1,y1] format.
-        df["bbox"] = df.bbox.apply(
-            lambda arr: [arr[0], arr[1], arr[0] + arr[2], arr[1] + arr[3]]
-        )
-        category_df = pd.DataFrame(instances_json["categories"]).rename(
-            {"id": "category_id"}, axis=1
-        )
-        annotations_df = df.merge(category_df, on="category_id")
+        """
+        Read instances and join to images
+        """
+        annotations_df = self._ann_to_df(instances_json)
         anno_df = (
             pd.DataFrame(
                 {
-                    "image_id": df.image_id,
+                    "image_id": annotations_df.image_id,
```

Comment:
is this the fix? while the rest are cleanups?

Reply:
yup, one line fix 🤦